tidalapi/page.py

```diff
@@ -19,54 +19,100 @@
 """
 
 import copy
-from typing import TYPE_CHECKING, Any, Dict, Iterator, List, Optional, Union, cast
+from dataclasses import dataclass
+from typing import (
+    TYPE_CHECKING,
+    Any,
+    Callable,
+    Dict,
+    Iterator,
+    List,
+    Optional,
+    Union,
+    cast,
+)
 
 from tidalapi.types import JsonObj
 
 if TYPE_CHECKING:
-    import tidalapi
+    from tidalapi.album import Album
+    from tidalapi.artist import Artist
+    from tidalapi.media import Track, Video
+    from tidalapi.mix import Mix
+    from tidalapi.playlist import Playlist, UserPlaylist
+    from tidalapi.request import Requests
+    from tidalapi.session import Session
 
+PageCategories = Union[
+    "Album",
+    "PageLinks",
+    "FeaturedItems",
+    "ItemList",
+    "TextBlock",
+    "LinkList",
+    "Mix",
+]
 
-class Page(object):
+AllCategories = Union["Artist", PageCategories]
+
+
+def _no_op_items(*args) -> List[Any]:
+    """A simple passthrough stub for category types that are not iterable."""
+    return []
+
+
+class Page:
     """
     A page from the https://listen.tidal.com/view/pages/ endpoint
 
     The :class:`categories` field will the most complete information
     However it is an iterable that goes through all the visible items on the page as well, in the natural reading order
     """
 
-    title = ""
-    categories: Optional[List[Any]] = None
-    _categories_iter: Optional[Iterator[Any]] = None
+    title: Optional[str] = ""
```

Comment:
Is it ever valid for the title to be unset?  why not just str?

Also more generally, do we *need* a default for these?  Lots of this code was written before typing exists, when the pattern was to set an empty sentinel on the class to indicate the spec.

See also https://github.com/tamland/python-tidal/issues/192 .  That's a bigger issue than here and I don't mind leaving the rest as optional for now, but I think this should just be `str`, as it *defaults* to a str.

Reply:
In [session.py](https://github.com/tamland/python-tidal/blob/master/tidalapi/session.py#L233) we initialize a `Page` with `None` as the `title`, however. If we're fine swapping that to an empty string, I'm fine stripping out the `Optional` here.

Reply:
I'd missed that.  don't really like it [requires null checking in everything downstream] but it's an implementation problem then, so keep it as optional.